docs/presentation/slides.md

```diff
@@ -0,0 +1,507 @@
+- Taking ownership of `FWDGT` gives us unique access to the peripheral registers, as the PAC only
+  allows one instance of each peripheral to exist. (It doesn't implement `Clone`.)
+- I've omitted `FreeWatchdog::setup` and some other methods which were too long to fit on the slide.
+- Type state pattern is used a fair bit for setting up peripherals.
+- Lots of (declarative) macros, for similar types and implementations where generics don't work.
+
+---
+
+# Initialisation
+
+```rust
+#[entry]
+fn main() -> ! {
+  let mut cp = cortex_m::Peripherals::take().unwrap();
+  let dp = pac::Peripherals::take().unwrap();
+
+  let mut rcu = dp.RCU.constrain();
+  let mut flash = dp.FMC.constrain();
+  let clocks = rcu.cfgr.sysclk(72.mhz()).adcclk(12.mhz()).freeze(&mut flash.ws);
+
+  let mut watchdog = FreeWatchdog::new(dp.FWDGT);
+  watchdog.start(WATCHDOG_MILLIS.ms());
+
+  let systick = SysTick::start(cp.SYST, &clocks);
+
+  let mut hoverboard = Hoverboard::new(dp.GPIOA, dp.TIMER0, &mut rcu.apb1,
+                                       clocks, ...);
+
+  // More initialisation...
+
+  loop {
+    watchdog.feed();
+
+    // Process events...
+  }
+}
+```
+
+???
+
+- `Peripherals::take()` returns a struct containing the PAC representations of all peripherals. If
+  you call it a second time it returns `None`, to prevent multiple references to the same registers
+  existing.
+- `constrain()` comes from extension traits which essentially convert the PAC type into the HAL
+  type.
+- `SysTick` gives us a clock, so we can measure time durations.
+
+---
+
+# Abstraction
+
+Let's make a wrapper for the peripherals on the hoverboard.
+
+```rust
+pub struct Hoverboard {
+  pub serial_rx: Rx<USART1>,
+  pub serial_writer: BufferedSerialWriter<Tx<USART1>>,
+  pub imu: Bmi160<I2cInterface<
+      BlockingI2c<I2C0, PB8<Alternate<AF1>>, PB9<Alternate<AF1>>>>>,
+  pub buzzer: Buzzer,
+  pub power_latch: PB2<Output<PushPull>>,
+  /// This will be high when the power button is pressed.
+  pub power_button: PC15<Input<Floating>>,
+  pub leds: Leds,
+  // More fields...
+}
+```
+
+???
+
+- `Bmi160` comes from another library, which gives us access to the IMU over I2C.
+- `Leds` is a struct with appropriately named fields for the various LEDs connected to GPIO pins.
+- `Buzzer` is a small wrapper around the PWM module from the HAL, which uses a timer to drive the
+  piezo speaker at a given frequency.
+- There are more fields for the other USART, charge state, and some configuration.
+- There is also a struct representing the motor, but it is stored in a static variable inside a
+  `Mutex` because it needs to be accessed from interrupt handlers.
+
+---
+
+# BLDC motor driver
+
+![BLDC motor driver circuit](bldc_driver.svg)
+
+???
+
+- Three motor windings connected between pairs of MOSFETs, like an H-bridge with an extra leg.
+- Each leg can be driven high, driven low, or undriven.
+
+---
+
+# BLDC motor control
+
+| Position | Sensor values | Forwards | Backwards |
```

Comment:
wow. This really doesn't render very nicely. Might be worth tweaking the CSS in index.html?

Reply:
I don't think this truth table makes much sense without a diagram of a wheel

Reply:
maybe just have columns of `sensor values -> position -> driving force`, describing how you might make it go forwards, and a footnote that to make it go in reverse, you do it backwards?

Reply:
Added some CSS and a diagram.